src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/CreateIncentive/CreateIncentiveCommandHandler.cs

```diff
@@ -7,28 +7,39 @@
 using System;
 using System.Threading;
 using System.Threading.Tasks;
-using SFA.DAS.EmployerIncentives.Enums;
+using Microsoft.EntityFrameworkCore.Internal;
+using SFA.DAS.EmployerIncentives.Commands.Types.IncentiveApplications;
 
 namespace SFA.DAS.EmployerIncentives.Commands.ApprenticeshipIncentive.CreateIncentive
 {
     public class CreateIncentiveCommandHandler : ICommandHandler<CreateIncentiveCommand>
     {
         private readonly IApprenticeshipIncentiveFactory _apprenticeshipIncentiveFactory;
         private readonly IApprenticeshipIncentiveDomainRepository _apprenticeshipIncentiveDomainRepository;
+        private readonly ICommandPublisher _commandPublisher;
 
         public CreateIncentiveCommandHandler(
             IApprenticeshipIncentiveFactory apprenticeshipIncentiveFactory,
-            IApprenticeshipIncentiveDomainRepository apprenticeshipIncentiveDomainRepository)
+            IApprenticeshipIncentiveDomainRepository apprenticeshipIncentiveDomainRepository,
+            ICommandPublisher commandPublisher)
         {
             _apprenticeshipIncentiveFactory = apprenticeshipIncentiveFactory;
             _apprenticeshipIncentiveDomainRepository = apprenticeshipIncentiveDomainRepository;
+            _commandPublisher = commandPublisher;
         }
 
         public async Task Handle(CreateIncentiveCommand command, CancellationToken cancellationToken = default)
         {
             var existing = await _apprenticeshipIncentiveDomainRepository.FindByApprenticeshipId(command.IncentiveApplicationApprenticeshipId);
             if (existing != null)
             {
+                if (existing.PendingPayments.Count > 0)
+                {
+                    var completeEarningCalculationCommand = new CompleteEarningsCalculationCommand(existing.Account.Id,
```

Comment:
This is breaking all our patterns.  Shouldn't the resilience check be looking for these scenarios/issues and publishing the CompleteEarningsCalculationCommand ?

Reply:
> This is breaking all our patterns. Shouldn't the resilience check be looking for these scenarios/issues and publishing the CompleteEarningsCalculationCommand ?
> 
> Or better handlethis in the SubmittedHandler event handler which can look up the incentive and Publish the CompleteEarningsCalculationCommand if needed else submit the CreateIncentiveCommand.

I'm not sure the SubmittedHandler would be better as it'd be crossing bounded contexts if it needs to get the ApprenticeshipIncentive.  Having a separate step in the resilience check would probably make sense tho.

Reply:
Yep, the purpose of the resilience check is to fix corrupt data so makes sense to add a step there to do just that.